UKHomeOffice/nmsw-ui
# NMSW-441 Add error mapping for invalid characters in FAL 1 files

# Ticket

NMSW-441

----

## AC

Given I have uploaded a FAL 1 file
When there are fields with invalid characters
Then I am shown an error message to help me fix them



Cell | Field | Rule | Error Message
-- | -- | -- | --
B3 | nameofship | contains character that is not alphanumeric | Enter the ship's name using English letters instead of special characters not recognised
F3 | callsign | contains characters that is not alphanumeric | Enter the call sign using only letters and numbers
B4 | signatory | contains character that is not alphanumeric | Enter the name of the master using English letters instead of special characters not recognised
B13 | cargo | alpha characters only | Enter the description using English letters instead of special characters not recognised


----

## To test

- upload a FAL 1 spreadsheet that has invalid characters in the fields listed above
- submit the file
- > you should see an error page with the above error messages
- > errors should be listed in row then column order (see image in Jira)

----

## Notes
